### PR TITLE
fix virtual repeat sizing based on unrendered items

### LINKED ISSUE #5999: Virtual repeat evaluates element size prior to compilation
This means, that, e.g. interpolated text is not accounted for in width/height calculations.

### PR DESCRIPTION
**Pull request by ashaffer**
Closes #5999.

I would like to have done something less blunt than a global `$apply` but i'm not sure of a better way to ensure I hit the right scope tree, since there's potentially a variety of complex scope relations going on here, and the trivial `block.scope.$digest()` does not seem to work, perhaps due to the sibling relationship of transcludes.

### CODE REVIEW

File: src/components/virtualRepeat/virtual-repeater.js [hunk 1]
```diff
@@ -497,7 +498,14 @@ VirtualRepeatController.prototype.readItemSize_ = function() {
 
   this.items = this.repeatListExpression(this.$scope);
   this.parentNode = this.$element[0].parentNode;
+
   var block = this.getBlock_(0);
+
+  // Make sure the block is fully constructed
+  if (!this.$rootScope.$$phase) {
+    this.$rootScope.$apply();
+  }
```
Comment (kseamon) [inline src/components/virtualRepeat/virtual-repeater.js]: This strikes me as the wrong approach. Forcing a digest of the block's scope should be sufficient.

Comment (ashaffer) [inline src/components/virtualRepeat/virtual-repeater.js]: Ya, as I mentioned, it seems like it should be, but it empirically is not.

**Comment by mackelito** [comment]
any updates on this issue?

**Comment by EladBezalel** [comment]
@ashaffer please rebase, squash your commits and follow our [Contribution Guide](https://github.com/angular/material/blob/master/CONTRIBUTING.md)

**Comment by ThomasBurleson** [comment]
@ashaffer - please rebase and we will merge!

**Comment by ashaffer** [comment]
Ah, sorry. Rebasing now.

**Comment by ashaffer** [comment]
Alright, rebased.

**Comment by ThomasBurleson** [comment]
@ashaffer - FYI, we will merge after the v1.1.0 release.

**Comment by ThomasBurleson** [comment]
@kseamon - please review.

File: src/components/virtualRepeat/virtual-repeater.js (hunk 1 above)
Comment (kseamon) [inline src/components/virtualRepeat/virtual-repeater.js]: How about digesting the parent scope of the directive?

**Comment by topherfangio** [comment]
@kseamon Ping for review please :-)

**Comment by kseamon** [comment]
I'd like an answer on that parent scope question, but other than that, lgtm

**Comment by ThomasBurleson** [comment]
@topherfangio - this has been in the queue for too long. Let's either research the parent scope, reject or lgtm.

**Comment by topherfangio** [comment]
@ThomasBurleson Associated issue was closed as deprecated. Let's do same here? I feel like there might be a better solution, and the issue didn't have any +1s.

**Comment by ThomasBurleson** [comment]
Will not merge.

**Pull request closed without merging**